### ISSUE TITLE
Add a rule to check for lock_timeout and statement_timeout

### ISSUE THREAD

**Issue by msnteixeira** [sbdchd/squawk#638]
Hey, I'd like to propose a new rule for the squawk linter to ensure that SQL migration queries include both `lock_timeout` and `statement_timeout` settings.

When running schema migrations on production databases, particularly for large tables, a query can be blocked by an existing lock, leading to a stalled migration and potential downtime. Similarly, a long-running statement can consume excessive resources. By enforcing the inclusion of these timeouts, we can prevent migrations from hanging indefinitely. This aligns with what is written in the squawk documentation [here](https://squawkhq.com/docs/safe_migrations#lock_timeout).

The linter should then check each SQL query within a migration file and raise a warning or error if it does not contain `SET lock_timeout` and `SET statement_timeout`. Here's an example:

Invalid SQL:
```sql
ALTER TABLE t ADD COLUMN c BOOLEAN;
Valid SQL:
```

Valid SQL:
```sql
SET lock_timeout = '1s';
SET statement_timeout = '5s';
ALTER TABLE t ADD COLUMN c BOOLEAN;
```

**Comment by joequreshi** [comment]
+1

**Comment by sbdchd** [comment]
Added a new rule to cover this in the latest version called `require-timeout-settings`.


```
❯ echo 'alter table t add column c text' | npx squawk-cli@latest
warning[require-timeout-settings]: Missing `set lock_timeout` before potentially slow operations
  ╭▸ stdin:1:1
  │
1 │ alter table t add column c text
  │ ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  │
  ├ help: Configure a `lock_timeout` before this statement.
  ╭╴
1 + set lock_timeout = '1s';
  ╰╴
warning[require-timeout-settings]: Missing `set statement_timeout` before potentially slow operations
  ╭▸ stdin:1:1
  │
1 │ alter table t add column c text
  │ ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  │
  ├ help: Configure a `statement_timeout` before this statement
  ╭╴
1 + set statement_timeout = '5s';
  ╰╴

Find detailed examples and solutions for each rule at https://squawkhq.com/docs/rules
Found 2 issues in 1 file (checked 1 source file)
```

Let me know what you think!
